whisper/whisperv6/filter.go

```diff
@@ -91,12 +99,49 @@ func (fs *Filters) Uninstall(id string) bool {
 	fs.mutex.Lock()
 	defer fs.mutex.Unlock()
 	if fs.watchers[id] != nil {
+		fs.removeFromTopicMatchers(fs.watchers[id])
 		delete(fs.watchers, id)
 		return true
 	}
 	return false
 }
 
+// addTopicMatcher adds a filter to the topic matchers
```

Comment:
addTopicMatcher adds a filter to the topic matchers. If the filter's `Topics` array is empty, it will be tried on every topic. Otherwise, it will be tried on the topics specified.